cli/cli.js

```diff
@@ -72,38 +76,124 @@ const runYoCommand = (cmd, args, options, opts) => {
+                    args = [];
+                    unknownArgs = [];
+                    let unknown = false;
+                    first.forEach(item => {
+                        if (item.startsWith('-')) {
+                            unknown = true;
+                        }
+                        if (unknown) {
+                            unknownArgs.push(item);
+                        } else {
+                            args.push(item);
+                        }
+                    });
+                } else if (first !== undefined) {
+                    args = [first];
+                }
+                cmdOptions = second;
+                if (third) {
+                    unknownArgs = (unknownArgs || []).concat(third);
+                }
+            } else {
+                // Option that doesn't contain arguments
+                // first=cmdOptions second=unknownArgs
+                args = [];
+                cmdOptions = first;
+                unknownArgs = second || [];
+            }
+
+            const firstUnknownArg = Array.isArray(unknownArgs) && unknownArgs.length > 0 ? unknownArgs[0] : undefined;
+            if (key === 'app' && firstUnknownArg !== undefined && !firstUnknownArg.startsWith('-')) {
+                // Unknown commands.
+                /* eslint-disable-next-line no-console */
+                console.error(
+                    chalk.red(`${chalk.yellow(firstUnknownArg)} is not a known command. See '${chalk.white(`${CLI_NAME} --help`)}'.`)
+                );
+
+                const cmd = Object.values(args).join('');
+                const availableCommands = program.commands.map(c => c._name);
+
+                const suggestion = didYouMean(cmd, availableCommands);
+                if (suggestion) {
+                    logger.info(`Did you mean ${chalk.yellow(suggestion)}?`);
+                }
+
+                process.exit(1);
+            }
+
+            // Get unknown options and parse.
+            const options = {
+                ...getCommandOptions(packageJson, unknownArgs),
+                ...addKebabCase(program.opts()),
+                ...addKebabCase(cmdOptions),
+            };
+
             if (opts.cliOnly) {
                 logger.debug('Executing CLI only script');
                 /* eslint-disable global-require, import/no-dynamic-require */
-                require(`./${key}`)(program.args, options, env);
+                require(`./${key}`)(args, options, env);
                 /* eslint-enable */
             } else {
                 const namespace = opts.blueprint ? `${packageNameToNamespace(opts.blueprint)}:${key}` : `${JHIPSTER_NS}:${key}`;
-                runYoCommand(namespace, program.args, options, opts);
+                runYoCommand(namespace, args, options, opts);
             }
         })
         .on('--help', () => {
             if (opts.help) {
-                logger.info(opts.help);
-            } else {
```

Comment:
The else was for getting help from the yo command, do you handle it somewhere else?

Reply:
It was not working.
I am build the help with commander.